From b542ad8784ca135d804f8ae3fd3f09ff21aa203b Mon Sep 17 00:00:00 2001
From: Ileana Dumitrescu <ileanadumitrescu95@gmail.com>
Date: Tue, 11 Jun 2024 17:58:47 +0300
Subject: [PATCH] libtool: Remove no-undefined for test with NetBSD

When compiling with no-undefined for NetBSD, the test fails,
so the flag has been removed for this OS.

* tests/flags.at: Remove no-undefined flag for NetBSD.
---
 tests/flags.at | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/tests/flags.at b/tests/flags.at
index fb4109db..e60b649c 100644
--- a/tests/flags.at
+++ b/tests/flags.at
@@ -26,7 +26,14 @@ m4_foreach([lt_tag], [CC, CXX, F77, FC, GCJ],
 AT_KEYWORDS([libtool])
 LT_AT_TAG(m4_defn([lt_tag]))
 
-LDFLAGS="$LDFLAGS -no-undefined"
+case $host_os in
+  netbsd*)
+    CXXFLAGS="$CXXFLAGS -shared-libgcc"
+    ;;
+  *)
+    LDFLAGS="$LDFLAGS -no-undefined"
+    ;;
+esac
 eval "`$LIBTOOL --config | $EGREP '^(FGREP)='`"
 
 m4_case(lt_tag,
-- 
2.45.1

